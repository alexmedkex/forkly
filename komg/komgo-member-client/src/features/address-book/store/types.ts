import { List } from 'immutable'
import { ImmutableMap } from '../../../utils/types'
import { Action } from 'redux'
import { IMember } from '../../members/store/types'

export enum AddressBookActionType {
  GET_COMPANIES_REQUEST = '@@addressBook/GET_COMPANIESS_REQUEST',
  GET_COMPANIES_SUCCESS = '@@addressBook/GET_COMPANIESS_SUCCESS',
  GET_COMPANIES_FAILURE = '@@addressBook/GET_COMPANIESS_FAILURE',
  GET_COMPANY_REQUEST = '@@addressBook/GET_COMPANY_REQUEST',
  GET_COMPANY_SUCCESS = '@@addressBook/GET_COMPANY_SUCCESS',
  GET_COMPANY_FAILURE = '@@addressBook/GET_COMPANY_FAILURE',
  GENERATE_MEMBER_PACKAGE_REQUEST = '@@addressBook/GENERATE_MEMBER_PACKAGE_REQUEST',
  GENERATE_MEMBER_PACKAGE_SUCCESS = '@@addressBook/GENERATE_MEMBER_PACKAGE_SUCCESS',
  GENERATE_MEMBER_PACKAGE_FAILURE = '@@addressBook/GENERATE_MEMBER_PACKAGE_FAILURE',
  ADD_COMPANY_TO_ENS_REQUEST = '@@addressBook/ADD_COMPANY_TO_ENS_REQUEST',
  ADD_COMPANY_TO_ENS_SUCCESS = '@@addressBook/ADD_COMPANY_TO_ENS_SUCCESS',
  ADD_COMPANY_TO_ENS_FAILURE = '@@addressBook/ADD_COMPANY_TO_ENS_FAILURE',
  CONFIGURE_MQ_REQUEST = '@@addressBook/CONFIGURE_MQ_REQUEST',
  CONFIGURE_MQ_SUCCESS = '@@addressBook/CONFIGURE_MQ_SUCCESS',
  CONFIGURE_MQ_FAILURE = '@@addressBook/CONFIGURE_MQ_FAILURE',
  CREATE_COMPANY_REQUEST = '@@addressBook/CREATE_COMPANY_REQUEST',
  CREATE_COMPANY_SUCCESS = '@@addressBook/CREATE_COMPANY_SUCCESS',
  CREATE_COMPANY_FAILURE = '@@addressBook/CREATE_COMPANY_FAILURE',
  UPDATE_COMPANY_REQUEST = '@@addressBook/UPDATE_COMPANY_REQUEST',
  UPDATE_COMPANY_SUCCESS = '@@addressBook/UPDATE_COMPANY_SUCCESS',
  UPDATE_COMPANY_FAILURE = '@@addressBook/UPDATE_COMPANY_FAILURE',
  DeleteCompanyByIdSuccess = '@@addressBook/DELETE_COMPANY_BY_ID_SUCCESS'
}

export interface AddressBookStateProperties {
  companies: List<IMember>
}

export type AddressBookState = ImmutableMap<AddressBookStateProperties>

export interface GetCompaniesRequest extends Action {
  type: AddressBookActionType.GET_COMPANIES_SUCCESS
  payload: IMember[]
}
export interface GetCompanyRequest extends Action {
  type: AddressBookActionType.GET_COMPANY_SUCCESS
  payload: IMember
}
export interface GenerateMemberPackageRequest extends Action {
  type: AddressBookActionType.GENERATE_MEMBER_PACKAGE_SUCCESS
  payload: IMember
}
export interface AddCompanyToENSRequest extends Action {
  type: AddressBookActionType.ADD_COMPANY_TO_ENS_SUCCESS
  payload: IMember
}
export interface ConfigureMQRequest extends Action {
  type: AddressBookActionType.CONFIGURE_MQ_SUCCESS
  payload: IMember
}

export interface CreateCompanyRequest extends Action {
  type: AddressBookActionType.CREATE_COMPANY_SUCCESS
  payload: IMember
}
export interface UpdateCompanyRequest extends Action {
  type: AddressBookActionType.UPDATE_COMPANY_SUCCESS
  payload: IMember
}

export interface DeleteCompanyByIdSuccess extends Action {
  type: AddressBookActionType.DeleteCompanyByIdSuccess
  payload: { id: string }
}

export type AddressBookActions =
  | GetCompaniesRequest
  | GetCompanyRequest
  | CreateCompanyRequest
  | UpdateCompanyRequest
  | GenerateMemberPackageRequest
  | AddCompanyToENSRequest
  | ConfigureMQRequest
  | DeleteCompanyByIdSuccess
