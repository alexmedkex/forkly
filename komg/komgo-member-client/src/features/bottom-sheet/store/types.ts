import { Action } from 'redux'
import { ImmutableMap } from '../../../utils/types'
import {
  CreateDocumentError,
  CreateDocumentSuccess,
  DocumentRegisteredError,
  DocumentRegisteredSuccess,
  HasId,
  HasName
} from '../../document-management/store/types'
import { ActionCreator } from 'react-redux'

export enum BottomSheetActionType {
  REMOVE_BOTTOMSHEET_ITEM = '@@btsh/REMOVE_BOTTOMSHEET_ITEM',
  RETRY_BOTTOMSHEET_ITEM = '@@btsh/RETRY_BOTTOMSHEET_ITEM',
  UPDATE_BOTTOMSHEET_ITEM = '@@btsh/UPDATE_BOTTOMSHEET_ITEM',
  CREATE_BOTTOMSHEET_ITEM = '@@btsh/CREATE_BOTTOMSHEET_ITEM'
}

export interface BottomSheetItem extends HasId, HasName {
  state: BottomSheetStatus
  itemType?: BottomSheetItemType
  displayStatus?: string
  action?: ActionCreator<any>
  navigationLink?: string
}

export enum BottomSheetItemType {
  REGISTER_KYC_DOCUMENT,
  CREATE_MAGIC_LINK,
  REGISTER_RD_DOCUMENT,
  REGISTER_TRADE_DOCUMENT
}

export enum BottomSheetStatus {
  NEW = 'NEW',
  PENDING = 'PENDING',
  REGISTERED = 'REGISTERED',
  FAILED = 'FAILED'
}

export interface BottomSheetStateFields {
  visible: boolean
  items: BottomSheetItem[]
}

export interface RemoveBottomSheetItem extends Action {
  type: BottomSheetActionType.REMOVE_BOTTOMSHEET_ITEM
  payload: HasId
}

export interface RetryRegisterItem extends Action {
  type: BottomSheetActionType.RETRY_BOTTOMSHEET_ITEM
  payload: BottomSheetItem
}

export interface UpdateBottomSheetItem extends Action {
  type: BottomSheetActionType.UPDATE_BOTTOMSHEET_ITEM
  payload: BottomSheetItem
}

export interface CreateBottomSheetItem extends Action {
  type: BottomSheetActionType.CREATE_BOTTOMSHEET_ITEM
  payload: BottomSheetItem
}

export type BottomSheetState = ImmutableMap<BottomSheetStateFields>

export type BottomSheetAction =
  | DocumentRegisteredSuccess
  | DocumentRegisteredError
  | CreateDocumentSuccess
  | CreateDocumentError
  | RemoveBottomSheetItem
  | RetryRegisterItem
  | UpdateBottomSheetItem
  | CreateBottomSheetItem
