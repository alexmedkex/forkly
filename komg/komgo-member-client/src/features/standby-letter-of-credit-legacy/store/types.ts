import { ImmutableMap, stringOrNull } from '../../../utils/types'
import { ActionType } from '../../letter-of-credit-legacy/store/types'
import { List, Map } from 'immutable'
import { IStandbyLetterOfCredit, TradeSource } from '@komgo/types/dist'

export enum StandbyLetterOfCreditActionType {
  SUBMIT_STANDBY_LETTER_OF_CREDIT_REQUEST = '@@standby-letters-of-credit/SUBMIT_LETTER_OF_CREDIT_REQUEST',
  SUBMIT_STANDBY_LETTER_OF_CREDIT_SUCCESS = '@@standby-letters-of-credit/SUBMIT_LETTER_OF_CREDIT_SUCCESS',
  SUBMIT_STANDBY_LETTER_OF_CREDIT_FAILURE = '@@standby-letters-of-credit/SUBMIT_LETTER_OF_CREDIT_FAILURE',
  GET_STANDBY_LETTER_OF_CREDIT_REQUEST = '@@standby-letters-of-credit/GET_STANDBY_LETTER_OF_CREDIT_REQUEST',
  GET_STANDBY_LETTER_OF_CREDIT_SUCCESS = '@@standby-letters-of-credit/GET_STANDBY_LETTER_OF_CREDIT_SUCCESS',
  GET_STANDBY_LETTER_OF_CREDIT_FAILURE = '@@standby-letters-of-credit/GET_STANDBY_LETTER_OF_CREDIT_FAILURE',
  FETCH_STANDBY_LETTERS_OF_CREDIT_REQUEST = '@@standby-letters-of-credit/FETCH_STANDBY_LETTERS_OF_CREDIT_REQUEST',
  FETCH_STANDBY_LETTERS_OF_CREDIT_SUCCESS = '@@standby-letters-of-credit/FETCH_STANDBY_LETTERS_OF_CREDIT_SUCCESS',
  FETCH_STANDBY_LETTERS_OF_CREDIT_FAILURE = '@@standby-letters-of-credit/FETCH_STANDBY_LETTERS_OF_CREDIT_FAILURE',
  FETCH_STANDBY_LETTER_OF_CREDIT_DOCUMENTS_REQUEST = '@@standby-letters-of-credit/FETCH_STANDBY_LETTER_OF_CREDIT_DOCUMENTS_REQUEST',
  FETCH_STANDBY_LETTER_OF_CREDIT_DOCUMENTS_SUCCESS = '@@standby-letters-of-credit/FETCH_STANDBY_LETTER_OF_CREDIT_DOCUMENTS_SUCCESS',
  FETCH_STANDBY_LETTER_OF_CREDIT_DOCUMENTS_FAILURE = '@@standby-letters-of-credit/FETCH_STANDBY_LETTER_OF_CREDIT_DOCUMENTS_FAILURE',
  ISSUE_STANDBY_LETTER_OF_CREDIT_REQUEST = '@@standby-letters-of-credit/ISSUE_STANDBY_LETTER_OF_CREDIT_REQUEST',
  ISSUE_STANDBY_LETTER_OF_CREDIT_SUCCESS = '@@standby-letters-of-credit/ISSUE_STANDBY_LETTER_OF_CREDIT_SUCCESS',
  ISSUE_STANDBY_LETTER_OF_CREDIT_FAILURE = '@@standby-letters-of-credit/ISSUE_STANDBY_LETTER_OF_CREDIT_FAILURE',
  REJECT_STANDBY_LETTER_OF_CREDIT_REQUEST = '@@standby-letters-of-credit/REJECT_STANDBY_LETTER_OF_CREDIT_REQUEST',
  REJECT_STANDBY_LETTER_OF_CREDIT_SUCCESS = '@@standby-letters-of-credit/REJECT_STANDBY_LETTER_OF_CREDIT_SUCCESS',
  REJECT_STANDBY_LETTER_OF_CREDIT_FAILURE = '@@standby-letters-of-credit/REJECT_STANDBY_LETTER_OF_CREDIT_FAILURE'
}

export interface ITradeId {
  source: TradeSource
  sourceId: string
}

export interface StandbyLetterOfCreditStateProperties {
  byId: Map<string, IStandbyLetterOfCredit>
  ids: List<string>
  total: number
}

export interface FetchStandByLettersOfCredit {
  filter: {
    options: any | undefined
  }
  polling?: boolean
}

export type StandbyLetterOfCreditState = ImmutableMap<StandbyLetterOfCreditStateProperties>
