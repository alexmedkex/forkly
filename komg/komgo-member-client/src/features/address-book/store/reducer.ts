import { Reducer } from 'redux'
import { List, Map } from 'immutable'
import { AddressBookActions, AddressBookActionType, AddressBookState, AddressBookStateProperties } from './types'
import { IMember } from '../../members/store/types'

const initialLicenseState: AddressBookStateProperties = {
  companies: List<IMember>()
}

export const initialState: AddressBookState = Map(initialLicenseState)

const onboardingReducer: Reducer<AddressBookState> = (
  state: AddressBookState = initialState,
  action: AddressBookActions
): AddressBookState => {
  switch (action.type) {
    case AddressBookActionType.GET_COMPANIES_SUCCESS:
      return state.set('companies', List<IMember>(action.payload))
    case AddressBookActionType.GENERATE_MEMBER_PACKAGE_SUCCESS:
    case AddressBookActionType.ADD_COMPANY_TO_ENS_SUCCESS:
    case AddressBookActionType.CONFIGURE_MQ_SUCCESS:
    case AddressBookActionType.GET_COMPANY_SUCCESS:
    case AddressBookActionType.UPDATE_COMPANY_SUCCESS:
      return state.update('companies', (members: List<IMember>) =>
        members.update(
          members.findIndex((member: IMember) => member.staticId === action.payload.staticId),
          () => action.payload
        )
      )
    case AddressBookActionType.CREATE_COMPANY_SUCCESS:
      return state.update('companies', (members: List<IMember>) => members.push(action.payload))

    case AddressBookActionType.DeleteCompanyByIdSuccess: {
      return state.update('companies', (members: List<IMember>) =>
        members.delete(members.findIndex((member: IMember) => member.staticId === action.payload.id))
      )
    }

    default:
      return state
  }
}

export default onboardingReducer
