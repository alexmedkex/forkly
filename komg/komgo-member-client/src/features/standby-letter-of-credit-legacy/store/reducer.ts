import { StandbyLetterOfCreditState, StandbyLetterOfCreditActionType } from './types'
import { fromJS, List, Map } from 'immutable'
import { Reducer, AnyAction } from 'redux'

export const initialState: StandbyLetterOfCreditState = fromJS({
  byId: Map(),
  ids: List(),
  total: 0
})

const reducer: Reducer<StandbyLetterOfCreditState> = (
  state: StandbyLetterOfCreditState = initialState,
  action: AnyAction
) => {
  switch (action.type) {
    case StandbyLetterOfCreditActionType.FETCH_STANDBY_LETTERS_OF_CREDIT_SUCCESS: {
      const ids = action.payload.items.map((letter: any) => letter.staticId)
      const update = action.payload.items.reduce(
        (memo: any, item: any) => ({
          ...memo,
          [item.staticId]: item
        }),
        {}
      )
      const standByLCs = state.get('byId').mergeDeep(fromJS(update))
      return state
        .set('byId', standByLCs)
        .set('ids', fromJS(ids))
        .set('total', action.payload.total)
    }
    case StandbyLetterOfCreditActionType.GET_STANDBY_LETTER_OF_CREDIT_SUCCESS: {
      const letterMap = state.get('byId').mergeDeep(
        fromJS({
          [action.payload.staticId]: action.payload
        })
      )

      return state.set('byId', letterMap)
    }
    default:
      return state
  }
}

export default reducer
