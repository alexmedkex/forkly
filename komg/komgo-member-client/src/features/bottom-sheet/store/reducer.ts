import * as immutable from 'immutable'
import { Reducer } from 'redux'

import {
  BottomSheetActionType,
  BottomSheetState,
  BottomSheetAction,
  BottomSheetStateFields,
  BottomSheetItem,
  BottomSheetStatus,
  BottomSheetItemType
} from './types'
import { DocumentActionType } from '../../document-management/store/'

const initialStateFields: BottomSheetStateFields = {
  visible: false,
  items: []
}

// @ts-ignore
export const initialState: BottomSheetState = immutable.Map(initialStateFields)

const reducer: Reducer<BottomSheetState> = (state = initialState, action: BottomSheetAction): BottomSheetState => {
  switch (action.type) {
    case BottomSheetActionType.CREATE_BOTTOMSHEET_ITEM:
    case DocumentActionType.CREATE_DOCUMENT_SUCCESS: {
      return createDocumentSuccessReducer(state, action)
    }
    case DocumentActionType.CREATE_DOCUMENT_ERROR: {
      return state
        .set('visible', true)
        .set('items', [
          pluckBottomSheetItems({ id: action.payload.id, name: action.payload.name, state: BottomSheetStatus.FAILED }),
          ...state.get('items').filter(item => item.id !== action.payload.id)
        ])
    }
    case BottomSheetActionType.UPDATE_BOTTOMSHEET_ITEM:
    case DocumentActionType.SHOW_DOCUMENT_REGISTERED_SUCCESS:
    case DocumentActionType.SHOW_DOCUMENT_REGISTERED_ERROR: {
      return updatingExistingItemReducer(state, action)
    }
    case BottomSheetActionType.REMOVE_BOTTOMSHEET_ITEM: {
      return removeBottomSheetItem(state, action)
    }
    case BottomSheetActionType.RETRY_BOTTOMSHEET_ITEM: {
      return state.set('visible', true).set('items', state.get('items').map(updateItemStatus(action.payload)))
    }
  }
  return state
}

export const removeBottomSheetItem = (state: BottomSheetState, action: BottomSheetAction) => {
  const items = state.get('items').filter(item => item.id !== action.payload.id)
  if (items.length === 0) {
    return state.set('items', items).set('visible', false)
  }
  return state.set('items', items)
}

export const updatingExistingItemReducer = (state: BottomSheetState, action: BottomSheetAction): BottomSheetState => {
  if (!verifyUpdatingExistingItem(action.payload.id, state.get('items'))) {
    return state
  }
  return state.set('visible', true).set(
    'items',
    state
      .get('items')
      .map(updateItemStatus(action.payload as BottomSheetItem))
      .reduce(moveUpdatedItemToFront(action.payload.id), [])
  )
}

export const verifyUpdatingExistingItem = (incomingId: string, items: BottomSheetItem[]): boolean => {
  if (!incomingId.length && items.length) {
    return false
  }
  return items.filter(item => item.id === incomingId).length > 0
}

export const createDocumentSuccessReducer = (state: BottomSheetState, action: BottomSheetAction): BottomSheetState => {
  // avoid duplicates (i.e, when retrying)
  const items = state.get('items').filter(item => item.id === action.payload.id)
  if (items.length > 0) {
    return state.set('visible', true).set(
      'items',
      state
        .get('items')
        .map(updateItemStatus(action.payload as BottomSheetItem))
        .reduce(moveUpdatedItemToFront(action.payload.id), [])
    )
  }

  return state
    .set('visible', true)
    .set('items', [
      pluckBottomSheetItems({ ...action.payload, itemType: maybeItemType(action.payload) }),
      ...state.get('items')
    ])
}

export const moveUpdatedItemToFront = (id: string) => (
  acc: BottomSheetItem[],
  item: BottomSheetItem
): BottomSheetItem[] => {
  return item.id === id ? [item, ...acc] : [...acc, item]
}

export function pluckBottomSheetItems(item: any): BottomSheetItem {
  return {
    id: item.id,
    state: item.state || BottomSheetStatus.PENDING,
    itemType: maybeItemType(item),
    name: item.name,
    displayStatus: item.displayStatus,
    action: item.action,
    navigationLink: item.navigationLink
  }
}

export const maybeItemType = (item: any) => {
  const { product } = item
  if (!product) {
    return undefined
  }
  switch (product.id) {
    case 'tradeFinance': {
      return item.context && item.context.subProductId === 'rd'
        ? BottomSheetItemType.REGISTER_RD_DOCUMENT
        : BottomSheetItemType.REGISTER_TRADE_DOCUMENT
    }
    case 'kyc':
      return BottomSheetItemType.REGISTER_KYC_DOCUMENT
  }
  return undefined
}

export const updateItemStatus = <T extends BottomSheetItem>(payload: T) => (item: BottomSheetItem): BottomSheetItem => {
  return item.id === payload.id
    ? ({ ...item, ...{ state: payload.state, displayStatus: payload.displayStatus } } as BottomSheetItem)
    : item
}

export default reducer
