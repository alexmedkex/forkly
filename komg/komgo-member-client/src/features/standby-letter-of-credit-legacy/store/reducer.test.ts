import { buildFakeStandByLetterOfCredit } from '@komgo/types'
import reducer, { initialState } from './reducer'
import { StandbyLetterOfCreditActionType } from './types'
import { fromJS } from 'immutable'

describe('StandByLetterOfCredit Reducer', () => {
  const standByLettersOfCredits = [
    buildFakeStandByLetterOfCredit({ staticId: '123' }),
    buildFakeStandByLetterOfCredit({ staticId: '1234' })
  ]

  it('should return default state', () => {
    const expected = initialState
    const invalidAction = { type: 'FOO', payload: ['bar'] }
    const actual = reducer(initialState, invalidAction)
    expect(actual).toEqual(expected)
  })

  describe('FETCH_STANDBY_LETTERS_OF_CREDIT_SUCCESS', () => {
    it('should store sblc in state when previous state was default', () => {
      const action = {
        type: StandbyLetterOfCreditActionType.FETCH_STANDBY_LETTERS_OF_CREDIT_SUCCESS,
        payload: { total: 2, items: standByLettersOfCredits, limit: 200, skip: 0 }
      }

      const expectedState = fromJS({
        byId: fromJS({
          [standByLettersOfCredits[0].staticId]: standByLettersOfCredits[0],
          [standByLettersOfCredits[1].staticId]: standByLettersOfCredits[1]
        }),
        ids: fromJS([standByLettersOfCredits[0].staticId, standByLettersOfCredits[1].staticId]),
        total: 2
      })

      const receivedState = reducer(initialState, action)

      expect(receivedState).toEqual(expectedState)
    })
    it('should match snapshot', () => {
      const action = {
        type: StandbyLetterOfCreditActionType.FETCH_STANDBY_LETTERS_OF_CREDIT_SUCCESS,
        payload: { total: 2, items: standByLettersOfCredits, limit: 200, skip: 0 }
      }
      const receivedState = reducer(initialState, action)
      expect(receivedState).toMatchSnapshot()
    })
  })

  describe('GET_STANDBY_LETTER_OF_CREDIT_SUCCESS', () => {
    it('should store sblc in state when previous state was default', () => {
      const action = {
        type: StandbyLetterOfCreditActionType.GET_STANDBY_LETTER_OF_CREDIT_SUCCESS,
        payload: standByLettersOfCredits[0]
      }

      const expectedState = fromJS({
        byId: fromJS({
          [standByLettersOfCredits[0].staticId]: standByLettersOfCredits[0]
        }),
        ids: [],
        total: 0
      })

      const receivedState = reducer(initialState, action)

      expect(receivedState).toEqual(expectedState)
    })
    it('should match snapshot', () => {
      const action = {
        type: StandbyLetterOfCreditActionType.GET_STANDBY_LETTER_OF_CREDIT_SUCCESS,
        payload: standByLettersOfCredits[0]
      }
      const receivedState = reducer(initialState, action)
      expect(receivedState).toMatchSnapshot()
    })
  })
})
