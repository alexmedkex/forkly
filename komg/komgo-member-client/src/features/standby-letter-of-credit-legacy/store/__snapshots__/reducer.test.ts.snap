// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StandByLetterOfCredit Reducer FETCH_STANDBY_LETTERS_OF_CREDIT_SUCCESS should match snapshot 1`] = `
Immutable.Map {
  "byId": Immutable.Map {
    "123": Immutable.Map {
      "cargoSnapshot": Immutable.Map {
        "parcels": Immutable.List [
          Immutable.Map {
            "deemedBLDate": undefined,
            "quantity": 600000,
            "loadingPlace": "Porto",
            "laycanPeriod": Immutable.Map {
              "startDate": "2017-12-31",
              "endDate": "2017-12-31",
            },
            "modeOfTransport": "VESSEL",
            "version": 2,
            "destinationPlace": "Weymouth",
            "_id": undefined,
            "id": "123",
          },
        ],
        "updatedAt": "2019-01-01T00:00:00.000Z",
        "status": "TO_BE_FINANCED",
        "sourceId": "E2389423",
        "cargoId": "F0401",
        "grade": "BRENT",
        "version": 2,
        "originOfGoods": "China",
        "_id": "123",
        "source": "KOMGO",
        "createdAt": "2019-01-01T00:00:00.000Z",
        "quality": "top",
      },
      "beneficiaryId": "0b5ad248-6159-47ca-9ac7-610c22877186",
      "contractDate": "2019-10-20",
      "expiryDate": "2019-10-11",
      "contractAddress": "0x2222",
      "overrideStandardTemplate": "Default legal template clauses",
      "stateHistory": Immutable.List [],
      "additionalInformation": "an additional info",
      "tradeSnapshot": Immutable.Map {
        "paymentTerms": Immutable.Map {
          "eventBase": "BL",
          "when": "AFTER",
          "time": 30,
          "timeUnit": "DAYS",
          "dayType": "CALENDAR",
        },
        "sellerEtrmId": undefined,
        "price": 70.02,
        "maxTolerance": 1.25,
        "deliveryLocation": "Weymouth",
        "creditRequirement": "DOCUMENTARY_LETTER_OF_CREDIT",
        "quantity": 600000,
        "priceUnit": "BBL",
        "commodity": "CRUDE_OIL",
        "updatedAt": "2019-01-01T00:00:00.000Z",
        "currency": "EUR",
        "status": "TO_BE_FINANCED",
        "buyer": "node.123",
        "deliveryTerms": "FOB",
        "deliveryPeriod": Immutable.Map {
          "startDate": "2019-04-03",
          "endDate": "2019-04-10",
        },
        "sourceId": "E2389423",
        "paymentTermsOption": "DEFERRED",
        "version": 1,
        "buyerEtrmId": "buyerEtrmId",
        "invoiceQuantity": "LOAD",
        "_id": "123",
        "dealDate": "2017-12-31",
        "source": "KOMGO",
        "createdAt": "2019-01-01T00:00:00.000Z",
        "minTolerance": 1.25,
        "seller": "node.321",
      },
      "staticId": "123",
      "documentHash": "0x0001",
      "reference": "REF-0000-00",
      "transactionHash": "0x1111",
      "feesPayableBy": "SPLIT",
      "issuingBankComment": "'Don't do that!",
      "issuingBankId": "1bc05a66-1eba-44f7-8f85-38204e4d3516",
      "updatedAt": "2019-01-01T00:00:00.000Z",
      "availableWith": "ISSUING",
      "nonce": 0,
      "currency": "USD",
      "status": "REQUESTED",
      "beneficiaryBankRole": "ADVISING_BANK",
      "issuingBankPostalAddress": "99-97 a street London EC1 000",
      "amount": 1000,
      "version": 1,
      "issuingBankReference": "REF-123",
      "contractReference": "CREF-123",
      "duplicateClause": "YES",
      "createdAt": "2019-01-01T00:00:00.000Z",
      "beneficiaryBankId": "a28b8dc3-8de9-4559-8ca1-272ccef52b47",
      "applicantId": "1d0d6806-6fe7-47b2-aef2-efd458d66f26",
      "tradeId": Immutable.Map {
        "source": "KOMGO",
        "sourceId": "55918db4-8a94-495c-a9b3-0aafebc74328",
      },
      "comment": "comment",
    },
    "1234": Immutable.Map {
      "cargoSnapshot": Immutable.Map {
        "parcels": Immutable.List [
          Immutable.Map {
            "deemedBLDate": undefined,
            "quantity": 600000,
            "loadingPlace": "Porto",
            "laycanPeriod": Immutable.Map {
              "startDate": "2017-12-31",
              "endDate": "2017-12-31",
            },
            "modeOfTransport": "VESSEL",
            "version": 2,
            "destinationPlace": "Weymouth",
            "_id": undefined,
            "id": "123",
          },
        ],
        "updatedAt": "2019-01-01T00:00:00.000Z",
        "status": "TO_BE_FINANCED",
        "sourceId": "E2389423",
        "cargoId": "F0401",
        "grade": "BRENT",
        "version": 2,
        "originOfGoods": "China",
        "_id": "123",
        "source": "KOMGO",
        "createdAt": "2019-01-01T00:00:00.000Z",
        "quality": "top",
      },
      "beneficiaryId": "0b5ad248-6159-47ca-9ac7-610c22877186",
      "contractDate": "2019-10-20",
      "expiryDate": "2019-10-11",
      "contractAddress": "0x2222",
      "overrideStandardTemplate": "Default legal template clauses",
      "stateHistory": Immutable.List [],
      "additionalInformation": "an additional info",
      "tradeSnapshot": Immutable.Map {
        "paymentTerms": Immutable.Map {
          "eventBase": "BL",
          "when": "AFTER",
          "time": 30,
          "timeUnit": "DAYS",
          "dayType": "CALENDAR",
        },
        "sellerEtrmId": undefined,
        "price": 70.02,
        "maxTolerance": 1.25,
        "deliveryLocation": "Weymouth",
        "creditRequirement": "DOCUMENTARY_LETTER_OF_CREDIT",
        "quantity": 600000,
        "priceUnit": "BBL",
        "commodity": "CRUDE_OIL",
        "updatedAt": "2019-01-01T00:00:00.000Z",
        "currency": "EUR",
        "status": "TO_BE_FINANCED",
        "buyer": "node.123",
        "deliveryTerms": "FOB",
        "deliveryPeriod": Immutable.Map {
          "startDate": "2019-04-03",
          "endDate": "2019-04-10",
        },
        "sourceId": "E2389423",
        "paymentTermsOption": "DEFERRED",
        "version": 1,
        "buyerEtrmId": "buyerEtrmId",
        "invoiceQuantity": "LOAD",
        "_id": "123",
        "dealDate": "2017-12-31",
        "source": "KOMGO",
        "createdAt": "2019-01-01T00:00:00.000Z",
        "minTolerance": 1.25,
        "seller": "node.321",
      },
      "staticId": "1234",
      "documentHash": "0x0001",
      "reference": "REF-0000-00",
      "transactionHash": "0x1111",
      "feesPayableBy": "SPLIT",
      "issuingBankComment": "'Don't do that!",
      "issuingBankId": "1bc05a66-1eba-44f7-8f85-38204e4d3516",
      "updatedAt": "2019-01-01T00:00:00.000Z",
      "availableWith": "ISSUING",
      "nonce": 0,
      "currency": "USD",
      "status": "REQUESTED",
      "beneficiaryBankRole": "ADVISING_BANK",
      "issuingBankPostalAddress": "99-97 a street London EC1 000",
      "amount": 1000,
      "version": 1,
      "issuingBankReference": "REF-123",
      "contractReference": "CREF-123",
      "duplicateClause": "YES",
      "createdAt": "2019-01-01T00:00:00.000Z",
      "beneficiaryBankId": "a28b8dc3-8de9-4559-8ca1-272ccef52b47",
      "applicantId": "1d0d6806-6fe7-47b2-aef2-efd458d66f26",
      "tradeId": Immutable.Map {
        "source": "KOMGO",
        "sourceId": "55918db4-8a94-495c-a9b3-0aafebc74328",
      },
      "comment": "comment",
    },
  },
  "ids": Immutable.List [
    "123",
    "1234",
  ],
  "total": 2,
}
`;

exports[`StandByLetterOfCredit Reducer GET_STANDBY_LETTER_OF_CREDIT_SUCCESS should match snapshot 1`] = `
Immutable.Map {
  "byId": Immutable.Map {
    "123": Immutable.Map {
      "cargoSnapshot": Immutable.Map {
        "parcels": Immutable.List [
          Immutable.Map {
            "deemedBLDate": undefined,
            "quantity": 600000,
            "loadingPlace": "Porto",
            "laycanPeriod": Immutable.Map {
              "startDate": "2017-12-31",
              "endDate": "2017-12-31",
            },
            "modeOfTransport": "VESSEL",
            "version": 2,
            "destinationPlace": "Weymouth",
            "_id": undefined,
            "id": "123",
          },
        ],
        "updatedAt": "2019-01-01T00:00:00.000Z",
        "status": "TO_BE_FINANCED",
        "sourceId": "E2389423",
        "cargoId": "F0401",
        "grade": "BRENT",
        "version": 2,
        "originOfGoods": "China",
        "_id": "123",
        "source": "KOMGO",
        "createdAt": "2019-01-01T00:00:00.000Z",
        "quality": "top",
      },
      "beneficiaryId": "0b5ad248-6159-47ca-9ac7-610c22877186",
      "contractDate": "2019-10-20",
      "expiryDate": "2019-10-11",
      "contractAddress": "0x2222",
      "overrideStandardTemplate": "Default legal template clauses",
      "stateHistory": Immutable.List [],
      "additionalInformation": "an additional info",
      "tradeSnapshot": Immutable.Map {
        "paymentTerms": Immutable.Map {
          "eventBase": "BL",
          "when": "AFTER",
          "time": 30,
          "timeUnit": "DAYS",
          "dayType": "CALENDAR",
        },
        "sellerEtrmId": undefined,
        "price": 70.02,
        "maxTolerance": 1.25,
        "deliveryLocation": "Weymouth",
        "creditRequirement": "DOCUMENTARY_LETTER_OF_CREDIT",
        "quantity": 600000,
        "priceUnit": "BBL",
        "commodity": "CRUDE_OIL",
        "updatedAt": "2019-01-01T00:00:00.000Z",
        "currency": "EUR",
        "status": "TO_BE_FINANCED",
        "buyer": "node.123",
        "deliveryTerms": "FOB",
        "deliveryPeriod": Immutable.Map {
          "startDate": "2019-04-03",
          "endDate": "2019-04-10",
        },
        "sourceId": "E2389423",
        "paymentTermsOption": "DEFERRED",
        "version": 1,
        "buyerEtrmId": "buyerEtrmId",
        "invoiceQuantity": "LOAD",
        "_id": "123",
        "dealDate": "2017-12-31",
        "source": "KOMGO",
        "createdAt": "2019-01-01T00:00:00.000Z",
        "minTolerance": 1.25,
        "seller": "node.321",
      },
      "staticId": "123",
      "documentHash": "0x0001",
      "reference": "REF-0000-00",
      "transactionHash": "0x1111",
      "feesPayableBy": "SPLIT",
      "issuingBankComment": "'Don't do that!",
      "issuingBankId": "1bc05a66-1eba-44f7-8f85-38204e4d3516",
      "updatedAt": "2019-01-01T00:00:00.000Z",
      "availableWith": "ISSUING",
      "nonce": 0,
      "currency": "USD",
      "status": "REQUESTED",
      "beneficiaryBankRole": "ADVISING_BANK",
      "issuingBankPostalAddress": "99-97 a street London EC1 000",
      "amount": 1000,
      "version": 1,
      "issuingBankReference": "REF-123",
      "contractReference": "CREF-123",
      "duplicateClause": "YES",
      "createdAt": "2019-01-01T00:00:00.000Z",
      "beneficiaryBankId": "a28b8dc3-8de9-4559-8ca1-272ccef52b47",
      "applicantId": "1d0d6806-6fe7-47b2-aef2-efd458d66f26",
      "tradeId": Immutable.Map {
        "source": "KOMGO",
        "sourceId": "55918db4-8a94-495c-a9b3-0aafebc74328",
      },
      "comment": "comment",
    },
  },
  "ids": Immutable.List [],
  "total": 0,
}
`;
